src/parser/core/modules/Defensives.tsx

```diff
@@ -1,89 +1,163 @@
 import {Data} from 'parser/core/modules/Data'
-import {Rows, TableStatistic, Statistics, SimpleStatistic} from 'parser/core/modules/Statistics'
-import React from 'react'
-import {AbstractStatisticOptions} from './Statistics/AbstractStatistic'
-
-export abstract class Defensives extends Analyser {
+import React, {Fragment, ReactNode} from 'react'
+import {Accordion, Button, Icon, Message, Table} from 'semantic-ui-react'
+import {CooldownHistoryEntry, Cooldowns} from './Cooldowns'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+import {Timeline} from './Timeline'
+
+const DEFENSIVE_ROLE_ACTIONS: Map<RoleKey, ActionKey[]> = new Map<RoleKey, ActionKey[]>([
+	['TANK', ['RAMPART', 'REPRISAL']],
+	['MELEE', ['FEINT', 'BLOODBATH', 'SECOND_WIND']],
+	['PHYSICAL_RANGED', ['SECOND_WIND']],
+	['MAGICAL_RANGED', ['ADDLE']],
+	['HEALER', []],
+])
+
+export class Defensives extends Analyser {
 	static override handle = 'defensives'
 	static override title = t('core.defensives.title')`Defensives`
+	static override displayOrder = DISPLAY_ORDER.DEFENSIVES
 
+	@dependency protected cooldowns!:Cooldowns
 	@dependency protected cooldownDowntime!: CooldownDowntime
 	@dependency protected data!: Data
-	@dependency private statistics!: Statistics
+	@dependency private timeline!: Timeline
 
 	/**
-	 * Implementing modules MUST provide a list of defensive actions to track
+	 * Implementing modules should provide a list of job-specific defensive actions to track
 	 */
-	protected abstract trackedDefensives: Action[]
-
+	protected trackedDefensives: Action[] = []
 	/**
-	 * Implementing modules may provide opts for the statistic display
+	 * Implementing modules may override the main header message text
 	 */
-	protected statisticOpts: AbstractStatisticOptions = {}
-
-	private uses: Map<Action['id'], number> = new Map()
+	protected headerContent: ReactNode = <Trans id="core.defensives.header.content">
+		Using your mitigation and healing cooldowns can help you survive mistakes, or relieve some stress on the healers and let them deal more damage.<br/>
+		While you shouldn't use them at the expense of your rotation or buff alignment, you should try to find helpful times to use them.
+	</Trans>
 
 	override initialise() {
-		const actionFilter = filter<Event>()
-			.source(this.parser.actor.id)
-			.type('action')
-
-		this.trackedDefensives.forEach(defensive => {
-			this.addEventHook(actionFilter.action(defensive.id),
-				() => this.uses.set(defensive.id, (this.uses.get(defensive.id) ?? 0) + 1)
-			)
+		const roleDefensives = DEFENSIVE_ROLE_ACTIONS.get(JOBS[this.parser.actor.job].role)?.map(key => this.data.actions[key]) ?? []
+		roleDefensives.forEach(roleAction => {
+			if (!this.trackedDefensives.find(action => roleAction.id === action.id)) {
+				this.trackedDefensives.push(roleAction)
+			}
 		})
+	}
 
-		this.addEventHook('complete', this.onComplete)
+	private getUses(defensive: Action): number {
+		return this.cooldowns.cooldownHistory(defensive).length
```

Comment:
This is failing to exclude interrupted casts and is instead counting them as uses.  See Pneuma on this log
http://localhost:3000/fflogs/A948DrvXpx1MzRaL/5/6

Reply:
Yeah, that's a thing. I updated the code to filter out any cooldown history entries where the cooldown ended due to interruption and that seems to have fixed the problem.